### ISSUE TITLE
Static code analyzer PHPStan

### ISSUE THREAD

**Issue by rugk** [PrivateBin/PrivateBin#816]
[PHPStan](https://phpstan.org/) looks like a good candidate/static code analysation tool, [and from the description it looks quite reasonable](https://phpstan.org/blog/find-bugs-in-your-code-without-writing-tests) – also with low false-positives, I hope. :upside_down_face: 

At least it claims to reduce the need to write unit tests… well… maybe. In any case, I guess it is not bad to introduce it.
It also supports PHPDoc, which I personally think is great as it improves type checking and prevents such errors.

**Comment by rugk** [comment]
Ah maybe [we should also enable PHP's strict types then](https://salferrarello.com/php-declare-strict_types-1/).

**Comment by elrido** [comment]
PHP `strict_types` seems to have been introduced with php 7. It will throw an error when you try to set it in PHP 5.

**Comment by rugk** [comment]
Okay, well… we thought about upgrading to PHP anyway (see https://github.com/PrivateBin/PrivateBin/issues/707#issuecomment-706938177), most distros only seem to maintain at at most of the end of 2021, only SLES until 2022, but we really don't need to care about each minor distro IMHO.

**Comment by elrido** [comment]
FYI, our currently used integration with Scrutinizer includes running PHPstan to generate it's reports.

**Comment by elrido** [comment]
> We [only support v7 and above](https://github.com/PrivateBin/PrivateBin/blob/b9d24b74b3f0fe5a379a830464db6926099f9533/.github/workflows/tests.yml#L27), so maybe this can be enabled now though? Or should I create a new issue for that?

Raised #1350 for that. We should maybe collect any items blocked by specific php versions still being supported in a specific project or other type of grouping, so we can easily find them once a certain php version is no longer supported by us.
